airflow/www/views.py

```diff
@@ -519,17 +519,23 @@ def last_dagruns(self, session=None):
     @has_access
     @provide_session
     def code(self, session=None):
-        dm = models.DagModel
-        dag_id = request.args.get('dag_id')
-        dag = session.query(dm).filter(dm.dag_id == dag_id).first()
+        all_errors = ""
+
         try:
-            with wwwutils.open_maybe_zipped(dag.fileloc, 'r') as f:
-                code = f.read()
+            dag_id = request.args.get('dag_id')
+            dag_orm = DagModel.get_dagmodel(dag_id, session=session)
+            dag = dag_orm.get_dag(STORE_SERIALIZED_DAGS)
```

Comment:
If you move the methods of reading code from DagModel to DagCode then you can avoid loading Dag here, because DagModel contains sufficient information.

Reply:
Yep, applied.

Reply:
Dag is needed to render the page anyway

Reply:
Why do we need DAG here? Is DagModel not enough? In the `dag_code.html` file we read only dag_id.

Reply:
Maybe we should introduce a new class DagCodeBag or something similar?

Reply:
The `dag` object passed to `self.render_template` comes from DagModel: Line522

so we don't need to get entire Dag here

Reply:
Ok, I shall pass dag model then and test if it works in webserver.